### ISSUE TITLE
terminal size != terminal size

### ISSUE THREAD

**Issue by lmorg** [kr/pty#51]
Hi

I'm having an issue where processes started via `pty` don't pick up my terminal dimensions. This causes issues with tools like `vi` and `top` (basically anything that draws a UI display based around the terminal size) to only use a subset of the parent terminal screen in the top left corner. It's particularly troublesome with tools that don't redraw the entire terminal, such as with `vi`.

Apologies for the vagueness of this issue, in truth I'm not even sure if it's an issue caused by `pty` specifically, but it certainly affects processes started with `pty`. The issue is I'm not really sure how terminal size is defined, let alone how to write the cgo / syscall code to change the setting.

Are you able to assist in any way?

(I'm not sure if this helps explain the problem, but this query relates to my $SHELL project: https://github.com/lmorg/murex)

**Comment by lmorg** [comment]
I worked out how to get around my issue. Turns out I dont need the this excellent package after all as I can just invoke the GO for standard library like so:

```
cmd := exec.Command(name, parameters...)

cmd.SysProcAttr = &syscall.SysProcAttr{
    Ctty: int(os.Stdout.Fd())
}

cmd.Stdin = os.Stdin
cmd.Stdout = os.Stdout
cmd.Stderr = os.Stderr
```

I couldn't have found this out without using the `pty` package though as it was only reading through your code that lead me towards the answer I was looking for.

**Comment by creack** [comment]
@lmorg did you see #24 ?

**Comment by lmorg** [comment]
I didn't, thank you for the link. Currently I found setting `SysProcAttr.Ctty` to the `os.Stdout.Fd()` seems to work in all cases I've tried (Linux, FreeBSD and OpenBSD) and since I'm forking a process inside the TTY (as I'm writing a $SHELL) rather than multiplexing, it makes much more sense to set the `Ctty` as described above. But I'll bare your package in mind should I find I need to do a bit of terminal management further down the line.
